### PR TITLE
Using controller-lifecycle-operator-sdk

### PR DESCRIPTION
**Pull request by jakub-dzon**
**What this PR does / why we need it**:
This PR makes use of new, common `controller-lifecycle-operator-sdk` library (https://github.com/kubevirt/controller-lifecycle-operator-sdk) that unifies common HCO-deployable operator code. 
The library has been built from CDI code responsible for managing deployment of CDI components (https://github.com/kubevirt/containerized-data-importer/tree/master/pkg/operator) and generalized enough to be usable by other operators (i.e. vm-import-operator - https://github.com/kubevirt/vm-import-operator/pull/380). 
Having that code in one place will ease development of new operators and maintenance of existing ones that eventually migrate to the new solution.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
None

**Release note**:
The intention of that PR was to be purely technical and to keep all of the CDI functionality unchanged.

```release-note
NONE
```

### CODE REVIEW

**Comment by jakub-dzon** [comment]
@pkliczewski, @mhenriks - please have a look,

**Comment by mhenriks** [comment]
Excellent work!

Before I get deep into the bits, I think it will be important to have a process in place for easy updates and releases/tags of `controller-lifecycle-operator-sdk`.  Also, I think there should probably a reference or dummy operator implementation in that repo with functional tests executed in CI.

**Comment by jakub-dzon** [comment]
Thanks, @mhenriks! 

I'm working on adding the reference operator implementation to the library and so far I've plugged the project into Travis CI so the unit tests get automatically executed.

As for the release process, I wanted to follow this (to be documented in the library project):
- A contributor creates PR or set of PRs;
- The PR(s) get reviewed and merged;
- Contributor creates issue with a request for release;
- Maintainer releases new version.

**Comment by jakub-dzon** [comment]
@mhenriks - I've added the reference operator implementation and functional tests in https://github.com/kubevirt/controller-lifecycle-operator-sdk/tree/master/examples/sample-operator. I have also posted a PR creating CI job to run these functional tests (https://github.com/kubevirt/project-infra/pull/628).

**Comment by mhenriks** [comment]
👍 really great stuff!  I think this will be ready to go once rebased

File: pkg/apiserver/webhooks/cdi-validate_test.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ package webhooks
 import (
 	"encoding/json"
 
+	sdkapi "github.com/kubevirt/controller-lifecycle-operator-sdk/pkg/sdk/api"
```
Comment (mhenriks) [inline pkg/apiserver/webhooks/cdi-validate_test.go:25]: why aren't these imports `kubevirt.io/controller-lifecycle-operator-sdk/pkg/sdk/api`?

Comment (mhenriks) [reply pkg/apiserver/webhooks/cdi-validate_test.go:25]: https://github.com/kubevirt/containerized-data-importer/issues/1380

To be implemented in follow up PR

**Pull request merged**